testsuite/cluster-test/src/cluster_swarm/cluster_swarm_kube.rs

```diff
@@ -241,14 +276,36 @@ impl ClusterSwarmKube {
                         }
                         bail!("job in still in progress {}", job_name)
                     }
-                    Err(e) => bail!("job_api.get failed for job {} : {:?}", job_name, e),
+                    Err(e) => {
+                        if killed {
+                            return Ok(true);
+                        }
+                        bail!("job_api.get failed for job {} : {:?}", job_name, e)
+                    }
                 }
             })
         })
         .await
     }
 
-    async fn run_jobs(&self, jobs: Vec<Job>, back_off_limit: u32) -> Result<()> {
+    pub async fn kill_job(&self, job_full_name: &str) -> Result<()> {
+        let dp = DeleteParams::default();
+        let job_api: Api<Job> = Api::namespaced(self.client.clone(), DEFAULT_NAMESPACE);
+        job_api
+            .delete(job_full_name, &dp)
+            .await?
+            .map_left(|o| println!("Deleting Job: {:?}", o.status))
+            .map_right(|s| println!("Deleted Job: {:?}", s));
+        let back_off_limit = 0;
+        // the job night have been deleted already, so we do not handle error
+        let _ = self
```

Comment:
I like the idea of supplying `killed` arg to not return error in wait_job_completion if we ourselves triggered kill

However I am a bit surprised we still need to ignore error even after having this arg.

I think we should not ignore error here - there is legitimate case for it, for example if job is not terminated for whatever reason [it is always good to surface some strange cases, instead of sweeping them under the rug, even if in theory you can argue that we don't care if job is deleted or not]

I think you might need to add another `if killed` to part where you handle status code, e.g. do not return error in `if let Some(failed) = job_status.failed` when `killed` is set

Reply:
I'll play around with your suggestion. My initial thought was that we're not guaranteed a good job_status code when we terminate a job, since that seems like it depends on the nature of the job we run.

Reply:
Updated to just print more logs. I'll still let it accept any job_status, but might be useful for debug later on.

```
DEBUG 2020-08-06 22:36:49 testsuite/cluster-test/src/cluster_swarm/cluster_swarm_kube.rs:471 Starting job run-forever for node ip-192-168-227-150.us-west-2.compute.internal
INFO 2020-08-06 22:36:49 testsuite/cluster-test/src/experiments/client_compatibility_test.rs:48 Wait for the other jobs to finish
Deleting Job: Some(JobStatus { active: Some(1), completion_time: None, conditions: None, failed: None, start_time: Some(Time(2020-08-06T22:36:49Z)), succeeded: None })
DEBUG 2020-08-06 22:37:09 common/retrier/src/lib.rs:43 job in still in progress run-forever-u6eywbvbyx. Retrying in 5 seconds..
INFO 2020-08-06 22:37:14 testsuite/cluster-test/src/cluster_swarm/cluster_swarm_kube.rs:281 Job run-forever-u6eywbvbyx has been killed already
INFO 2020-08-06 22:37:14 testsuite/cluster-test/src/cluster_swarm/cluster_swarm_kube.rs:306 Killing job run-forever-u6eywbvbyx returned job_status.success
INFO 2020-08-06 22:37:14 testsuite/cluster-test/src/main.rs:640 Experiment finished, waiting until all affected validators recover
```